pmxbot/slack.py

```diff
@@ -86,27 +92,25 @@ def handle_message(self, msg):
             return
         nick = resolve_nick(msg)
 
-        channel = self.slack.server.channels.find(msg['channel']).name
-        channel = core.AugmentableMessage(channel, thread=msg.get('thread_ts'))
+        channel_name = (
+            self.slacker.conversations.info(msg['channel'])
+            .body.get('channel', {})
+            .get('name')
```

Comment:
This code looks potentially overly-conservative. Will it work the same with `.body['channel']['name']`? I would be surprised if passing `None` for the `channel_name` would work as expected.

Reply:
There is a valid case where a message doesn't have a name, which is when you're talking directly to pmxbot (an IM)
<img width="1062" alt="image" src="https://user-images.githubusercontent.com/1598192/198834190-59249c87-0483-43e9-aced-93d3704488c6.png">

As also seen in the example from the docs: https://api.slack.com/methods/conversations.list#:~:text=%22dGVhbTpDMDYxRkE1UEI%3D%22%0A%20%20%20%20%7D%0A%7D-,Alternate%20response,-Example%20response%20when

In this case I think we'd want to continue with a channel of `None` so that the actions still function, since we'll only need the channel_id.

So `name` isn't always present, but I think its fair to assume that `channel` is always present in the body if the request is successful so I will change to:

`self.slacker.conversations.info(msg['channel']).body['channel'].get('name')`

... but will also adapt so that we're not using `slacker` as you suggested in another comment.

Reply:
Thanks for the investigation and explanation.